src/pages/filter/ui/panels/function-panel/function-panel.store.ts

```diff
@@ -0,0 +1,75 @@
+import { makeAutoObservable } from 'mobx'
+
+import datasetStore from '@store/dataset'
+import filterStore from '@store/filter'
+
+class FunctionPanelStore {
+  constructor() {
+    makeAutoObservable(this)
+  }
+
+  public get variants(): string[] {
+    const { variants } = filterStore.statFuncData
+    return variants
+  }
+
+  public get filterName(): string {
+    return filterStore.selectedGroupItem.name
+  }
+
+  public get filterGroup(): string {
+    return filterStore.selectedGroupItem.vgroup
+  }
+
+  public getCachedValues<T>(componentName: string): T {
+    return filterStore.readFilterCondition(componentName) as T
+  }
+
+  public setCachedValues<T>(componentName: string, conditions: T): void {
+    filterStore.setFilterCondition<T>(componentName, conditions)
+  }
+
+  public clearCachedValues(componentName: string): void {
+    filterStore.clearFilterCondition(componentName)
+  }
+
+  public async applyConditions<T>(
+    componentName: string,
+    conditions: T,
+  ): Promise<void> {
+    await datasetStore.setConditionsAsync([['func', componentName, conditions]])
+  }
+
+  public addSelectedFilters(variant: [string, number]): void {
+    filterStore.addSelectedFilters({
+      group: this.filterGroup,
+      groupItemName: this.filterName,
+      variant: variant,
+    })
+  }
+
+  public async handleSumbitConditionsAsync<T>(
```

Comment:
looks like overengineering, better to pass locusValue here and create conditions inside this function
No need to be async as well